pkg/operator/operator.go

```diff
@@ -0,0 +1,287 @@
+	"context"
+	"crypto/x509"
+	"flag"
+	"io/ioutil"
+	"os"
+	"time"
+	"fmt"
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	nodeobservabilityv1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha1"
+	nodeobservabilityv1alpha2 "github.com/openshift/node-observability-operator/api/v1alpha2"
+	operatorconfig "github.com/openshift/node-observability-operator/pkg/operator/config"
+	caconfigmapcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/ca-configmap"
+	machineconfigcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig"
+	nodeobservabilityrun "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservabilityrun"
+	nodeobservabilitycontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservability"
+)
+
+const (
+	operatorName = "node_observability_operator"
+)
+
+var (
+	setupLog = ctrl.Log.WithName("operator")
+)
+
+// Client holds the API clients required by Operator.
+type Client struct {
+	client.Client
+	meta.RESTMapper
+}
+
+// Operator hold the client, manager, and logging resources
+// for the nodeobservability opreator.
+type Operator struct {
+	client  Client
+	manager manager.Manager
+	log     logr.Logger
+}
+
+// Aggregate kubebuilder RBAC tags in one location for simplicity.
+// +kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilityes,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilityes/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilityes/finalizers,verbs=update
+// +kubebuilder:rbac:groups=cloudcredential.openshift.io,resources=credentialsrequests;credentialsrequests/status;credentialsrequests/finalizers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=route.openshift.io,resources=routes,verbs=get;watch;list
+// +kubebuilder:rbac:groups=config.openshift.io,resources=infrastructures,verbs=get;list;watch
+// local role
+// +kubebuilder:rbac:groups="",namespace=node-observability-operator,resources=secrets;serviceaccounts;configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="apps",namespace=node-observability-operator,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",namespace=node-observability-operator,resources=pods,verbs=get;list;watch
```

Comment:
That's why the RBAC is messed up. Pay attention to what you copy from the other operators.